specs/Print.md

```diff
@@ -0,0 +1,473 @@
+    CHECK_FEATURE_RETURN(webviewEnvironment6);
+
+    wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+    CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+    wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+    CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+    CHECK_FEATURE_RETURN(printSettings2);
+
+    CHECK_FAILURE(printSettings2->put_Orientation(COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT));
+    CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(true));
+
+    CHECK_FAILURE(
+        webView2_15->PrintToPdfStream(
+            printSettings2.get(),
+            Callback<ICoreWebView2PrintToPdfStreamCompletedHandler>(
+                [this](HRESULT errorCode, IStream* pdfData) -> HRESULT
+                {
+                    CHECK_FAILURE(errorCode);
+                    DisplayPdfDataInPrintDialog(pdfData);
+                    return S_OK;
+                })
+                .Get()));
+     return true;
+}
+```
+
+```c#
+async void PrintToPdfStream()
+{
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings();
+    printSettings.Orientation = CoreWebView2PrintOrientation.Portrait;
+    printSettings.ShouldPrintBackgrounds = true;
+
+    MemoryStream pdfStream = new MemoryStream();
+    System.IO.Stream stream = await webView.CoreWebView2.PrintToPdfStreamAsync(printSettings);
+    DisplayPdfDataInPrintDialog(pdfData);
+}
+
+// Function to display current page pdf data in a custom print preview dialog.
+void DisplayPdfDataInPrintDialog(Stream pdfData)
+{
+    // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+/// Specifies the print dialog kind.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_DIALOG_KIND {
+  /// Opens the browser print preview dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW,
+
+  /// Opens the system print dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_SYSTEM_PRINT,
+
+} COREWEBVIEW2_PRINT_DIALOG_KIND;
+
+/// Specifies the printing side.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_SIDE {
+  /// Single-sided printing.
+  COREWEBVIEW2_PRINT_SIDE_SINGLE,
+
+  /// Double-sided, horizontal printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_HORIZONTAL,
+
+  /// Double-sided, vertical printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_VERTICAL,
+
+} COREWEBVIEW2_PRINT_SIDE;
+
+[uuid(92269C88-1CC1-47E0-9A87-71458409BA15), object, pointer_default(unique)]
+interface ICoreWebView2_15 : ICoreWebView2_14 {
+  /// Print the current page asynchronously to the specified printer with the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. The handler will return `E_ABORT`
+  /// if `printerName` doesn't match with the name of any printers on the OS
+  /// or invalid settings for a given printer.
```

Comment:
A reply later in the PR notes that if you ask for color, but the printer doesn't support color, then you get black-and-white. On the other hand, this comment says that if you pass invalid settings, then nothing prints at all.

So what counts as an "invalid" setting? If I ask for collation or two-sided printing but the printer doesn't support that feature, is that "invalid" or merely "not supported (best effort result)"?

Reply:
Good question!

Invalid settings here are those values that doesn't exist in printer capabilities (DevCaps). In our API implementation, we check against installed printers on the OS, their capabilities and if provided values are not in printer capabilities, we invoke the completed handler with an error.

I think we should also validate the color/collate/two-side property with the printer cap and the value doesn't exist, we return the handler with an error. 

Eg., if ask for color printing from a B/W printer, return handler with error.

Note: Native Print API doesn't throw any error if we pass two-sided printing for a printer that only supports single-sided.